src/parsers.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  Config,
+  isBooleanConfig,
+  isCustomConfig,
+  isNumberConfig,
+  isStringConfig,
+  isStringEnumConfig,
+  ResolvedConfigValue,
+} from "./types";
+
+function parseString(value: unknown): string {
+  if (typeof value !== "string") {
+    throw new Error("not a string");
+  }
+  return value;
+}
+
+function parseNumber(value: unknown): number {
+  if (typeof value === "number" && Number.isFinite(value)) {
+    return value;
+  }
+  if (typeof value === "string") {
+    if (!Number.isFinite(parseFloat(value))) {
+      throw new Error("not a number");
+    }
+    return parseFloat(value);
+  }
+  throw new Error("not a number");
+}
+
+function parseBoolean(value: unknown): boolean {
+  if (typeof value === "boolean") {
+    return value;
+  }
+  if (typeof value === "string" && ["true", "false"].includes(value.toLowerCase())) {
```

Comment:
Good logic 👍  Is there a test for it?

Reply:
https://github.com/contiamo/react-runtime-config/blob/a333f6599d254b18df2d82343a8f3a273fdd7199/src/index.test.ts#L243-L255